spec/ssh_scan/rule_engine_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rspec'
+require 'ssh_scan/rule_engine'
+
+describe SSHScan::RuleEngine do
+  context "for conditionals" do
+    it "eq, neq should work for any literal" do
+      result = {:ssh_lib => "openssh"}
+      expect(SSHScan::RuleEngine.match({"ssh_lib_eq" => "openssh"}, result)).to be true
+      expect(SSHScan::RuleEngine.match({"ssh_lib_neq" => "dropbear"}, result)).to be true
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.match({"ssh_version_eq" => 1.99}, result)).to be false
+      expect(SSHScan::RuleEngine.match({"ssh_version_neq" => 1.99}, result)).to be true
+    end
+
+    it "gt, lt, gte, lte should work for numeric literal" do
+      result = {:ssh_version => 2.0}
+      expect(SSHScan::RuleEngine.match({"ssh_version_gt" => 1.99}, result)).to be true
+      expect(SSHScan::RuleEngine.match({"ssh_version_gte" => 2.0}, result)).to be true
+      expect(SSHScan::RuleEngine.match({"ssh_version_lt" => 1.99}, result)).to be false
+      expect(SSHScan::RuleEngine.match({"ssh_version_lte" => 2.0}, result)).to be true
```

Comment:
Recommend testing below/eql/above examples (to verify boundaries) for each verb, like so...

    # Testing gt boundaries
    expect(SSHScan::RuleEngine.match({"ssh_version_gt" => 1.99}, result)).to be true
    expect(SSHScan::RuleEngine.match({"ssh_version_gt" => 2.0}, result)).to be false
    expect(SSHScan::RuleEngine.match({"ssh_version_gt" => 2.1}, result)).to be false

    # Testing gte boundaries
    expect(SSHScan::RuleEngine.match({"ssh_version_gte" => 1.99}, result)).to be true
    expect(SSHScan::RuleEngine.match({"ssh_version_gte" => 2.0}, result)).to be true
    expect(SSHScan::RuleEngine.match({"ssh_version_gte" => 2.1}, result)).to be false

    # Testing lt boundaries
    expect(SSHScan::RuleEngine.match({"ssh_version_lt" => 1.99}, result)).to be false
    expect(SSHScan::RuleEngine.match({"ssh_version_lt" => 2.0}, result)).to be false
    expect(SSHScan::RuleEngine.match({"ssh_version_lt" => 2.1}, result)).to be true

    # Testing lte boundaries
    expect(SSHScan::RuleEngine.match({"ssh_version_lte" => 1.99}, result)).to be false
    expect(SSHScan::RuleEngine.match({"ssh_version_lte" => 2.0}, result)).to be true
    expect(SSHScan::RuleEngine.match({"ssh_version_lte" => 2.1}, result)).to be true